src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/ClientWebSocket.cs

```diff
@@ -0,0 +1,563 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
Rather than entirely re-implementing ClientWebSocket, can you instead just create a WebSocketHandle.Browser.cs ala the WebSocketHandle.Managed.cs at https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Managed.cs ?  This "handle" is how we handled (no pun intended) the cross-platformness in the past here, with a WebSocketHandle.WinHttp.cs and a WebSocketHandle.WinRT.cs in previous releases.  We no longer have those because they now just use the managed one, but the abstraction remains in place for need like the one in this PR.

Reply:
Hey Stephen.  

We are discussing the same thing and seeing if it actually fits.